src/main/kotlin/no/nav/helse/domene/arbeid/domain/Arbeidsforhold.kt

```diff
@@ -17,11 +17,25 @@ sealed class Arbeidsforhold(open val arbeidsgiver: Virksomhet, open val startdat
                             val permisjon: List<Permisjon> = emptyList(),
                             val arbeidsavtaler: List<Arbeidsavtale> = emptyList()): Arbeidsforhold(arbeidsgiver, startdato, sluttdato) {
 
-        fun yrke() = gjeldendeArbeidsavtale().yrke
+        val gjeldendeArbeidsavtale: Arbeidsavtale.Gjeldende
 
-        fun gjeldendeArbeidsavtale() = arbeidsavtaler.first {
-            it.tom == null
+        init {
+            if (arbeidsavtaler.isEmpty()) {
+                throw IllegalArgumentException("et arbeidsforhold må ha minst én arbeidsavtale")
+            }
+
+            val antallGjeldendeArbeidsavtaler = arbeidsavtaler.filter { it is Arbeidsavtale.Gjeldende }.size
+
+            if (antallGjeldendeArbeidsavtaler == 0) {
+                throw IllegalArgumentException("et arbeidsforhold må ha én gjeldende arbeidsavtale")
+            } else if (antallGjeldendeArbeidsavtaler > 1) {
+                throw IllegalArgumentException("et arbeidsforhold kan ikke ha flere gjeldende arbeidsavtaler")
+            }
```

Comment:
Unødvendig med to tester når du kun er interessert i om antallGjeldendeArbeidsavtaler != 1

Reply:
Enig, men det er interessant å sjekke om påstanden `et arbeidsforhold må ha én gjeldende arbeidsavtale` holder. Dersom begge tilfellene logges med samme feilmelding, så er det umulig å finne ut av (med mindre det telles på andre måter)

Reply:
"et arbeidsforhold skal ha én gjeldende avtale, denne har ${antallGjeldendeArbeidsavtaler.size}" ?

Reply:
det høres enklere ut. tok med den oppforbi i endringen også